pkg/apps/strategy/support/lifecycle_pod_hook.go

```diff
@@ -38,28 +45,30 @@ type lifecyclePodHook struct {
 	hookCompleteChan chan struct{}
 	hookFailureError error
 
-	// logStreamStarted indicates that log streaming from the hook pod container has already been started
-	logStreamStarted bool
+	// logStreamCompleted indicates that log streaming from the hook pod container has already been started
+	logStreamCompleted bool
 
 	// logStreamCompleteChan signals that log stream from the hook pod container has been finished
 	logStreamCompleteChan chan struct{}
 
 	out io.Writer
+
+	workqueue workqueue.RateLimitingInterface
```

Comment:
i switched this to workqueue to guarantee that there is just one item being processed at the time... i guess using just informers with resource event handlers directly will lead into flakes/race as they might run asynchronously (?).